Source/Core/Core/AchievementManager.cpp

```diff
@@ -987,6 +994,38 @@ void AchievementManager::FetchBadge(AchievementManager::Badge* badge, u32 badge_
       if (name_to_fetch.empty())
         return;
     }
+
+    std::string cache_base = File::GetUserPath(D_RETROACHIEVEMENTSCACHE_IDX);
+    auto context = Common::SHA1::CreateContext();
+    UpdateFromArray(*context, Common::BitCastToArray<u8>(badge_type));
+    context->Update(std::vector<u8>(name_to_fetch.begin(), name_to_fetch.end()));
+    auto name_hash = context->Finish();
+    std::string filename;
+    for (size_t ix = 0; ix < name_hash.size(); ix++)
+    {
+      u8 upper = name_hash[ix] / 16;
+      filename.append(1, upper + ((upper > 9) ? 'A' - 10 : '0'));
+      u8 lower = name_hash[ix] % 16;
+      filename.append(1, lower + ((lower > 9) ? 'A' - 10 : '0'));
+    }
```

Comment:
And this should be a utility function so you can actually reuse it. `Common::SHA1::ToString(const Common::SHA1::Digest& digest)` or whatever.

Reply:
How opposed would you be to this and UpdateFromArray going in SHA1.h/cpp?

Reply:
Yes please, do that.